server/src/internalClusterTest/java/org/elasticsearch/indices/settings/UpdateNumberOfReplicasIT.java

```diff
@@ -440,10 +439,7 @@ public void testUpdateWithInvalidNumberOfReplicas() {
 
     public void testUpdateNumberOfReplicasAllowNoIndices() {
         createIndex("test-index", Settings.builder().put("index.number_of_replicas", 0).build());
-        final IndicesOptions options = new IndicesOptions(
-            EnumSet.of(IndicesOptions.Option.ALLOW_NO_INDICES),
-            EnumSet.of(IndicesOptions.WildcardStates.OPEN)
-        );
+        final IndicesOptions options = IndicesOptions.DEFAULT;
```

Comment:
I looked into this:

- This test was using the flags `IndicesOptions.Option.ALLOW_NO_INDICES` and `IndicesOptions.WildcardStates.OPEN`, these meant that it will expand only to open indices and it will allow empty wildcard expressions.
- The default of `UpdateSettingsRequest` does not match that: https://github.com/elastic/elasticsearch/blob/9f2d38856d3f689344fe768e10a59a5eb91438c4/server/src/main/java/org/elasticsearch/action/admin/indices/settings/put/UpdateSettingsRequest.java#L43
Let me decipher it, it expands to both open and closed indices and it does not allow empty expressions.

If I remove this I am going to change the indices options the test receives.

Reply:
Looking into this test, I do not fully understand its purpose and I completely agree with you, that you would think that it shouldn't be setting any special parameters. 

According to the name, the intention is to test that `update numbe-of-replicas allows (expressions that match) no indices`; this passes because the flag `ALLOW_NO_INDICES` was set during this test using the multi-syntax default. So, it's not testing that every `update of number-of-replicas allows (expressions that match) no indices` but merely that if you set it like this it will work. 

Shouldn't we be testing the default behaviour?